### ISSUE TITLE
[crash] I jumped into the Fischer...ouch 

### ISSUE THREAD

**Issue by TheQuadShot** [BrassGoggledCoders/SteamCraft2#282]
This is when I jumped in and it was generating the dim.
http://pastebin.com/FGwVz5TM

This is when I tried to log back into the world

http://pastebin.com/SbR7YBAw

**Comment by warlordjones** [comment]
Oh noes, this crash is back again :(

**Comment by TheQuadShot** [comment]
yup, and I really durped, I thought I did a quick backup before i jumped, but it didn't finish, so I lost most of what I did today... at least I got all those resources back! lol

**Comment by warlordjones** [comment]
D: D: D: Oh noes....

**Comment by warlordjones** [comment]
Should theoretically be fixed...

**Comment by vagaprime** [comment]
I am still getting the same crash with the newest version of the mod. here is a copy of my crash log

http://pastebin.com/z0dSQu27

**Comment by warlordjones** [comment]
Aww, nuts. I'll keep trying to squish this...

**Comment by vagaprime** [comment]
Ya any world that crashes with this bug crashes anytime you try try to launch it. Is there a way to disable your dim? I am planning on including your mod in my next release of my pack on FTB and I would like to keep world breaking crashes to a minimum.

**Comment by warlordjones** [comment]
You can disable the generation of the portal to the dimension in the generation.cfg (within the sc2 folder in config)

**Comment by vagaprime** [comment]
Thank you

**Comment by Febilian** [comment]
I would like to point out that this issue also affects server-based worlds when they generate for the first time; I'd put this mod(among others) on a private server to test some stuff in a multiplayer environment, and the first time the world was being initially loaded, it appeared to attempt to load part of the Deeps(as well as other dimension files, whether the Nether/End or added by other mods) ahead of time - and caused this error + crash when generating the world files for the first time, and then anytime after that in which the world is loaded by the server.

To be specific, it was the error prefaced with "java.lang.NullPointerException: Exception getting block type in world", as seen in the first log posted originally for this issue. Sadly I do not believe I have the log itself anymore, as this happened a little while ago, and I assumed the issue was likely fixed since it did not occur in a new world I created after updating to one of the Jenkins builds.

I suppose I was able to bypass it for the time being by just generating a new world and getting lucky in the sense that the Deeps area it tried to generate beforehand did not have whatever quirk causes the crash. I'm not any expert on coding, but based on the crash report, it appears to be a generation based problem specific to the dimension(which you probably already knew, granted) that isn't consistently showing itself because it only happens under specific circumstances while generating the Deeps. If you haven't already, it might be worth double-checking the generation code for the Deeps, how the different generated blocks interact with each other, maybe even checking the code for the blocks that generate in the Deeps or the code that implements them into the biome gen, and creating multiple different worlds(preferably in a multiplayer/server based environment, as it seems to try to load it regardless of whether it's been entered or not) and entering the Deeps in them to test whether the crash occurs.

A way to disable the dimension completely would really help for now, if simply disabling the portal blocks does not actually stop the Deeps itself from generating in the world files in a server environment then disabling the portal would do nothing to help.

**Comment by warlordjones** [comment]
Yeah, the issue is that I can't seem to reproduce it at ALL in development environments, which is a real pain.

I will add a way to disable the dimension registration as soon as possible

**Comment by Febilian** [comment]
This might be a long shot, but is it possible that another mod that people install without really giving it a second thought might be messing with the Deeps generation somehow? Something that by default affects most standard dimensions and isn't treating the Deeps as an exception and causing issues? I don't know if you're testing the mod all on its lonesome or not, and a lot of people use more than one mod when they play in a modded environment.

**Comment by warlordjones** [comment]
I use a couple of mods, but it's plausible its one I don't use. I'll check the sets logs for common denominators when I have more time (which is not now, after I add this cfg I need to sleep :P)

**Comment by Febilian** [comment]
Fair enough! If it happens to me again, I'll try to pick out which mods might be commonly used in the set I'm running(or just send the whole list over, at least all the mods that would be installed both clientside and serverside). Also I'll actually save the log in such an event. Hopefully that might help a bit more.

**Comment by warlordjones** [comment]
Thanks!

**Comment by warlordjones** [comment]
@Febilian The config to disable the registration should be available in the latest jenkins build. It's in generation.cfg :) And with that, goodnight!

**Comment by TheDarkPreacher** [comment]
looks like that crash is still there. 

http://pastebin.com/WLBCgKmD

made a backup, hopped in, crashed during dimension load, tried to get back in, crashed again. going to disable the Deeps in my next version of my modpack, Steam Engineering.

**Comment by TheDarkPreacher** [comment]
well, it would seem that The Deeps just needs to be disabled completely, or possibly make the portal blocks disabled through a config option.

I would hate to have to take this mod out of my pack for the next release. I disabled Deeps gen through the config (generation.cfg) and got the following crash after trying to hop into the portal.

http://pastebin.com/88XqMf4A

**Comment by warlordjones** [comment]
You can disable the portals if you use the jenkins release available here: http://jenkins.k-4u.nl/view/BrassGoggledCoders/job/SteamCraft%202/ Make sure to grab a jenkins boilerplate: http://jenkins.k-4u.nl/view/BrassGoggledCoders/job/Boilerplate/ I should have a stable release with this option out soon, but I'd recommend using this in the meantime.
